src/model.rs

```diff
@@ -159,6 +160,21 @@ impl Collections {
             }
         }
 
+        fn dedup_collection<T: Clone + Eq + Hash>(source: &mut Collection<T>) -> Collection<T> {
```

Comment:
Maybe with `vec.dedup`, `vec.dedup_by` to avoid cloning

Reply:
but we must also manage Idx<T>

Reply:
I'd propose this implementation instead for the entire function.

```rust
use std::iter::FromIterator;
let set: HashSet<T> = std::collections::HashSet::from_iter(source.take());
Collection::new(set.into_iter().collect())
```
It should avoid reallocation of all the `T`. One of the downside is that we allocate a `HashSet` but I think we can't really avoid that (unless implementing ourselves the one-to-one comparison to identify the duplicates, but I don't think it worth it). A second downside is the allocation of the `Vec` (with the `.collect()`) but this second one could be avoided if we implement `FromIterator` on `Collection` (needs a modification of `typed_index_collection`).

Reply:
Maybe not HashSet for stability.

Reply:
> Maybe not HashSet for stability.

Sure... but implementing `PartialOrd` on the objects `T` might not always make sense.

We might also take a look at [`Vec::dedup()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup). The downside seems to be that it would work only for sorted `Vec`... and sorting relies on `PartialOrd` :facepalm: